### PR TITLE
WIP: add feedback to batch processor

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

Extends batch-processor to be able to send feedback messages.
Feedback messages must be provided in the "raw" format.

Works both with gRPC and REST.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Nov 16 18:32:18 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=2

**Comment by seldondev** [comment]
Mon Nov 16 18:32:26 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=1

**Comment by seldondev** [comment]
Wed Nov 18 15:17:05 UTC 2020
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=4

**Comment by seldondev** [comment]
Wed Nov 18 15:17:26 UTC 2020
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=3

**Comment by seldondev** [comment]
Wed Nov 18 15:22:29 UTC 2020
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=5

**Comment by seldondev** [comment]
Wed Nov 18 15:22:29 UTC 2020
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=6

File: python/seldon_core/batch_processor.py [hunk 1]
```diff
@@ -270,6 +293,86 @@ def _send_batch_predict(
...
+    feedback_kwargs = {}
+    meta = {
+        "tags": {
+            "batch_id": batch_id,
+            "batch_instance_id": batch_instance_id,
+            "batch_index": batch_idx,
+        }
+    }
+    # Feedback Protos does not support meta and SeldonClient.feedback does not support headers
+    # feedback_kwargs["meta"] = meta
```
Comment (axsaucedo) [inline python/seldon_core/batch_processor.py:356]: Remove (or if moved to single function can be left under if statement)

File: python/seldon_core/batch_processor.py [hunk 2]
```diff
@@ -270,6 +293,86 @@ def _send_batch_predict(
...
+    # feedback_kwargs["headers"] = {"Seldon-Puid": batch_instance_id}
+    try:
+        data = json.loads(input_raw)
+        feedback_kwargs["raw_request"] = data
+
+        str_output = None
+        for i in range(retries):
+            try:
+                # TODO: Add functionality for explainer
```
Comment (axsaucedo) [inline python/seldon_core/batch_processor.py:365]: Move todos into issues where possiuble

**Review by axsaucedo — COMMENTED** [review]
Looks good! Added a couple of minor comments, other than that looks good to go

File: python/seldon_core/batch_processor.py [hunk 3]
```diff
@@ -270,6 +293,86 @@ def _send_batch_predict(
                 # TODO: Optimize client to share session for requests
                 seldon_payload = sc.predict(**predict_kwargs)
                 assert seldon_payload.success
+                str_output = json.dumps(seldon_payload.response)
+                break
+            except (requests.exceptions.RequestException, AssertionError) as e:
+                # print("Exception:", e, "retries:", retries)
```
Comment (axsaucedo) [inline python/seldon_core/batch_processor.py:299]: REmove

**Comment by seldondev** [comment]
Wed Nov 18 15:39:13 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=7

**Comment by seldondev** [comment]
Wed Nov 18 15:39:18 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=8

**Comment by seldondev** [comment]
Wed Nov 18 15:42:40 UTC 2020
The logs for [lint] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=10

**Comment by seldondev** [comment]
Wed Nov 18 15:42:41 UTC 2020
The logs for [pr-build] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=9

**Comment by seldondev** [comment]
Wed Nov 18 15:50:22 UTC 2020
The logs for [lint] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=12

**Comment by seldondev** [comment]
Wed Nov 18 15:50:28 UTC 2020
The logs for [pr-build] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2653/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2653 --build=11

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/2653#issuecomment-729769496 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-729769496/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**